panic: runtime error: index out of range

Function `parseListLine` panics when called with the argument "000000000x " (notice the space at the end).

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/jlaffaye/ftp.parseHostedFTPLine(0x10e3042, 0xb, 0x114c200, 0xc4200107d0, 0xc420010701)
	/Users/Metalnem/Go/src/github.com/jlaffaye/ftp/parse.go:196 +0x24a
github.com/jlaffaye/ftp.parseListLine(0x10e3042, 0xb, 0xc4200107d0, 0x0, 0x0)
	/Users/Metalnem/Go/src/github.com/jlaffaye/ftp/parse.go:209 +0x68
github.com/jlaffaye/ftp.SuperParseLine(0x10e3042, 0xb, 0xc420014630, 0x29, 0xc420014630)
	/Users/Metalnem/Go/src/github.com/jlaffaye/ftp/parse.go:218 +0x35
main.main()
	/Users/Metalnem/Go/src/github.com/jlaffaye/ftp/main/main.go:12 +0x3a
exit status 2
```

This bug was found with the help of [go-fuzz](https://github.com/dvyukov/go-fuzz).

--- a/parse.go
+++ b/parse.go
@@ -192,14 +192,14 @@ func parseHostedFTPLine(line string) (*Entry, error) {
 	}
 
 	scanner := newScanner(line)
-	fields := scanner.NextFields(9)
-	if fields[1] == "0" { // Set link count to 1 and attempt to parse as Unix.
-		fields[1] = "1"
-		newLine := strings.Join(fields, " ")
-		return parseLsListLine(newLine)
-	}
-	return nil, errUnsupportedListLine
+	fields := scanner.NextFields(2)
 
+	if len(fields) < 2 || fields[1] != "0" {
+		return nil, errUnsupportedListLine
+	}
+
+	// Set link count to 1 and attempt to parse as Unix.
+	return parseLsListLine(fields[0] + " 1 " + scanner.Remaining())
 }
 
 // parseListLine parses the various non-standard format returned by the LIST

--- a/parse_test.go
+++ b/parse_test.go
@@ -69,6 +69,7 @@ var listTestsFail = []unsupportedLine{
 	{"modify=20150806235817;invalid;UNIX.owner=0; movies", "Unsupported LIST line"},
 	{"Zrwxrwxrwx   1 root     other          7 Jan 25 00:17 bin -> usr/bin", "Unknown entry type"},
 	{"total 1", "Unsupported LIST line"},
+	{"000000000x ", "Unsupported LIST line"}, // see https://github.com/jlaffaye/ftp/issues/97
 	{"", "Unsupported LIST line"},
 }
 

